### ISSUE TITLE
Question about gradient checking of batch norm

### ISSUE THREAD

**Issue by arnaldog12** [harrybraviner/batch_norm_python#1]
Hi, @harrybraviner 

Lately, I have been implementing a neural network with batch normalization from scratch with python and numpy only. I followed the formulas from original paper and the guides from these tutorials:
- https://wiseodd.github.io/techblog/2016/07/04/batchnorm/
- https://kevinzakka.github.io/2016/09/14/batch_normalization/

Actually, my problem is that the gradients do not check when I perform gradient checking. I found your implementation and saw that you wrote the following in the documentation of your gradient checking function:
 > """
It's possible that some gradient components will be 
and others won't be. In fact, it's a lot easier to get the '
(e.g. self._out_W) gradients correct that the 'earlier' 
Therefore this function breaks down the checking into named chunks.
"""

But I did not understand it completely. Why may some gradients not check? Why are later gradients easier to be correct than earlier ones?

Thank you in advance
